ota-plus-web/app/com/advancedtelematic/api/ApiClient.scala

```diff
@@ -241,6 +242,8 @@ class UserProfileApi(val conf: Configuration, val apiExec: ApiClientExec)(implic
     userProfileRequest(s"${namespace.get}/credentials/$keyUuid")
       .transform(_.withMethod("GET"))
       .execResult(apiExec)
+
+  private def segment(userId: UserId) = URLEncoder.encode(userId.id, "UTF-8")
```

Comment:
I would suggest to move the encoding logic to https://github.com/advancedtelematic/ota-plus-server/blob/master/ota-plus-web/app/com/advancedtelematic/api/ApiClient.scala#L49 and https://github.com/advancedtelematic/ota-plus-server/blob/master/ota-plus-web/app/com/advancedtelematic/api/ApiClient.scala#L57 and encode the whole path there

Reply:
I had thought about that as well, but it would mean to parse the url, encode only the segments and putting it back together again. (Otherwise ":" and "/" would also get encoded.) That didn't seem better to me.

Reply:
Damn... 
we will need to find some solution later. I can imagine we will get this problem again in the future...

Reply:
I agree. I haven't found what the actual issue is yet, unfortunately.